src/sentry/api/method/definition.py

```diff
@@ -0,0 +1,35 @@
+from functools import wraps
+
+
+def sentry_api():
+    """
+    This decorator handles the API overheads like validating and deserializing inputs,
+    serializing outputs, paginating lists, and more feature to be added in the future.
+
+    It works with the SentrySchema to document everything around the endpoint.
+
+    ```
+    class ApiEndpoint(Endpoint):
+
+        @sentry_api(
+            public=True,
+            query_params=[Serializer1, Serializer2],
+            response_format=Serializer3,
+            paginated=True
+        )
+        def get(self, request, serializer1_obj, serialzer2_obj):
```

Comment:
Why do the serializers get passed here?

Reply:
The idea is the wrapper would validate the data, and pass in the validated data objects.
so i guess `serializer1_dict` would be better

Reply:
`deserialized_obj` instead?

Reply:
a single deserializer obj, or one for each serializer?

Reply:
Hmm, maybe `serializer1_validated_data` actually? Kind of wordy though

Reply:
So in my hackweek project, i had it so the Serializer can define the argument name after deserialization. I was thinking of keeping that going.
so
```
deserialized_name = "validated_args"
```
becomes
```
def get(self, request, validated_args):
```

Reply:
Hmm, that seems like maybe it should be defined on the api level, like

`query_params={"project": ProjectSerializer}` etc

I guess it's a bit more boilerplate, but also gives you a better idea of where those params are coming from when reading the endpoint. Otherwise it might be a bit hard to understand (especially for new folks). One other benefit is that if we ever had a case of needing to use the same serializer twice in an api then we'd support that too.